File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/autovalue/Week.java
package com.tickaroo.tikxml.annotationprocessing.textcontent.autovalue;

import com.google.auto.value.AutoValue;
import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
@Xml
@AutoValue
public abstract class Week {

  @Element
  public abstract List<Day> days();
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/autovalue/Week.kt
package com.tickaroo.tikxml.annotationprocessing.textcontent.autovalue

import com.google.auto.value.AutoValue
import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
@AutoValue
abstract class Week {
    @Element
    abstract fun days(): List<Day>
}
